src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollectionOfT.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Represents an <see cref="IEdmObject"/> that is a collection of <see cref="IEdmChangedObject"/>s.
+    /// </summary>
+    [NonValidatingParameterBinding]
+    public class EdmChangedObjectCollection<TStructuralType> : EdmChangedObjectCollection, ICollection<IEdmChangedObject<TStructuralType>>, IEdmObject
+    {
+        private Collection<IEdmChangedObject<TStructuralType>> _items;
```

Comment:
>Collection [](start = 16, length = 10)

Why Collection rather than ICollection? Was there something you needed in Collection that was not in ICollection?